src/debug/debugConfigurationProvider.ts

```diff
@@ -0,0 +1,58 @@
+import * as vscode from "vscode";
+import * as nls from "vscode-nls";
+
+nls.config({ messageFormat: nls.MessageFormat.bundle, bundleFormat: nls.BundleFormat.standalone })();
+const localize: nls.LocalizeFunc = nls.loadMessageBundle();
+
+export class DebugConfigurationProvider implements vscode.DebugConfigurationProvider {
+    resolveDebugConfiguration(_folder: vscode.WorkspaceFolder | undefined, debugConfiguration: vscode.DebugConfiguration, _token?: vscode.CancellationToken | undefined): vscode.ProviderResult<vscode.DebugConfiguration> {
+        if (!debugConfiguration.type && !debugConfiguration.request && !debugConfiguration.name) {
```

Comment:
Does this let you use the active editor as a CMake script playground without manually writing a launch configuration? That's pretty cool.

Reply:
Yes! That's exactly right. This allows you to click the "Run and Debug" button when there are no launch configurations. 

![DebugCMakeScript](https://github.com/microsoft/vscode-cmake-tools/assets/86264750/b264db2d-3707-40a6-b3fa-b9d80a3fb22c)

Now that you mention it, it actually gives me another idea. This entry point isn't available, to my knowledge, if you do have launch configurations. So, I could add either a better entry point in the debug dropdown, or even a context menu entry point on files that is present only in .cmake files. I might look into this before merging.

Reply:
![CMake_AutoDebugConfig](https://github.com/microsoft/vscode-cmake-tools/assets/86264750/d1cabced-a8e9-48e3-8dd7-9989f43afa5e)